### PR TITLE
Maven 3.6.3 as minimum requirements for plugins

### PR DESCRIPTION
**Pull request by slawekjaranowski**
References:

https://github.com/apache/maven-site/commit/e1a353dfd98f0cc716fa3ad45409ed109072f9b0 #398 
https://github.com/apache/maven-site/commit/5cb62af4d232ee06c1fe2239b81cd06fc4b02501 #410 

https://github.com/apache/maven-site/commit/9f3a79c441837dd01e2b00cddf578ac54d492a8f 

https://lists.apache.org/thread/k2g3cy4hfktg159ckx0vk9hygrotnj11

### CODE REVIEW

File: content/apt/developers/compatibility-plan.apt [hunk 1]
```diff
@@ -50,7 +50,7 @@ Maven Plugins Compatibility Plan
 
   * Since June 2020, Maven Plugin API used by plugins >= 3.1.0 + Java 8 prerequisites {{{https://s.apache.org/MVN-PLUGIN-MIGRATION-3.1}Technical details}}
 
-  * In March 2023, Maven before 3.2.5 has been {{{/docs/history.html#maven-before-3-2-5}marked as EOL}}, meaning that plugins should use 3.2.5 as minimum
+  * In April 2023, Maven before 3.6.3 has been {{{/docs/history.html#maven-3-6-x-and-before}marked as EOL}}, meaning that plugins should use 3.6.3 as minimum
```
Comment (michael-o) [inline content/apt/developers/compatibility-plan.apt:53]: If you retain the information above, you should also retain the previous 3.2.5, a changelog so to speak.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: content/apt/developers/compatibility-plan.apt (hunk 1 above)
Comment (slawekjaranowski) [reply content/apt/developers/compatibility-plan.apt:53]: I can preserve both lines .... but link `/docs/history.html#maven-before-3-2-5` is not valid now.
It was only a moth between such changes.

Comment (michael-o) [reply content/apt/developers/compatibility-plan.apt:53]: I'd say that we need a chronologically reserve changelog. That will be understood and people will see what happened over time.

Comment (slawekjaranowski) [reply content/apt/developers/compatibility-plan.apt:55]: Please look now.

Comment (michael-o) [reply content/apt/developers/compatibility-plan.apt:55]: April 2024

**Review by michael-o — CHANGES_REQUESTED** [review]

File: content/apt/developers/compatibility-plan.apt (hunk 1 above)
Comment (slawekjaranowski) [reply content/apt/developers/compatibility-plan.apt:55]: April `2024` will be in next three months. Now we have January `2024`

Comment (michael-o) [reply content/apt/developers/compatibility-plan.apt:55]: My bad, misunderstanding.

**Review by michael-o — APPROVED** [review]

**Review by slachiewicz — APPROVED** [review]

**Pull request merged**